tools/hlt_client/hlt_client/download_game.py

```diff
@@ -125,32 +126,40 @@ def __init__(self, destination, date, all_bots=False):
 
 class UserGameDownloader(GameDownloader):
     _USER_BOT_URI = 'https://api.halite.io/v1/api/user/{}/match?limit={}&offset={}'
+    _DATE_BOT_URI = 'https://api.halite.io/v1/api/user/{}/match?limit={}&offset={}&filter=time_played,>=,{}'
     _FETCH_THRESHOLD = 250
     _BUCKETS = []
 
-    def __init__(self, destination, user_id, limit):
+    def __init__(self, destination, user_id, limit, date):
         """
         Download games for a user
         :param destination: Where to download
         :param user_id: Which user's replays to fetch
         :param limit: How many replays to fetch (max)
+        :param date: Which date to download
         """
         self.destination = destination
-        self.objects = self._parse_user_metadata(self._fetch_metadata(user_id, limit))
+        self.objects = self._parse_user_metadata(self._fetch_metadata(user_id, limit, date))
 
-    def _fetch_metadata(self, user_id, limit):
+    def _fetch_metadata(self, user_id, limit, date):
         """
         Retrieves paginated game metadata from the halite servers for a specified user up to limit items
         :param user_id: The id of the user to fetch
         :param limit: The maximum number of items to fetch
+        :param date: Which date to download
         :return: The full metadata of items
         """
         print('Fetching Metadata')
         current = 0
         result_set = []
         while current <= limit:
             current_limit = self._FETCH_THRESHOLD if ((limit - current) >= self._FETCH_THRESHOLD) else (limit - current)
-            result_set += requests.get(self._USER_BOT_URI.format(user_id, current_limit, current)).json()
+            if date is None:
+                result_set += requests.get(self._USER_BOT_URI.format(user_id, current_limit, current)).json()
+            else:
+                requested_date = datetime.datetime.strptime(date,'%Y%m%d').strftime('%Y-%m-%dT00:00')
```

Comment:
Btw, if you're always going to filter at the level of a full day you can drop the `T00:00`.

Reply:
Right, I just dropped the T00:00.